lisa/analysis/rta.py

```diff
@@ -0,0 +1,339 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2018, Arm Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from numpy import unique
+
+from lisa.utils import memoized
+from lisa.analysis.base import AnalysisHelpers
+from lisa.analysis.base import TraceAnalysisBase
+from lisa.trace import requires_events
+
+class Events(TraceAnalysisBase):
+    """
+    Support for RTA events analysis.
+
+    :param trace: input Trace object
+    :type trace: :class:`trace.Trace`
+    """
+
+    name = 'rta'
+
+    def __init__(self, trace):
+        """
+        Load peformance data of an rt-app workload
+        """
+        logger = self.get_logger()
+
+        self.trace = trace
+
+        if 'rtapp_task' not in self.trace.base_trace.available_events:
+            raise ValueError('No rt-app tasks events found in trace')
+
+        tasks_df = self.trace.base_trace.df_events('rtapp_task')
+        self.rtapp_tasks = unique(tasks_df["__comm"].values).tolist()
```

Comment:
To ease interaction with the rest of LISA, it might be a good idea to store a list of TaskID instead of plain names:
```
        tasks = <a list of PID or comms or tuples (PID, comm)>
        task_ids = [trace.get_task_id(task, update=False) for task in tasks]
        # Filtering the dataframe is handled by lisa.datautils.df_filter_task_ids.
        # The columns it uses can be changed, if needed.
        df = df_filter_task_ids(df, task_ids)
```

https://lisa-linux-integrated-system-analysis.readthedocs.io/en/master/misc_utilities.html?highlight=df_filter#lisa.datautils.df_filter_task_ids

Reply:
Ok, I did now something along those lines.